documentation/DPG-quickstart.md

```diff
@@ -0,0 +1,288 @@
+
+Please refer to this [link](https://github.com/Azure/azure-sdk-for-js/blob/main/CONTRIBUTING.md#prerequisites) for the environment set up prerequisites in azure-sdk-for-js repository.
+
+
+# Project folder structure and name convention
+
+If you are the first time to prepare the SDK, please follow the Azure SDK guidance and discuss with architects to decide the project folder and name convention for DPG libraries.
+
+1. SDK Repo Root.
+   The generated libraries should be in the [azure-sdk-for-js](https://github.com/Azure/azure-sdk-for-js) repo, so fork and clone it in your local then the absolute path is called **${SDK_REPO_ROOT} folder**.
+
+1. Project Folder Structure.  
+   Normally, the folder structure would be something like `sdk/{servicename}/{servicename}-{modulename}`. For example, we have `sdk/agrifood/agrifood-farming` folder for Farmbeats account modules under {SDK_REPO_ROOT}. That folder will be your **${PROJECT_ROOT} folder**.  
+1. Package Name Convention.  
+   The package name for DPG is something like `@azure/{servicename}-{modulename}`. For example, the package name for Farmbeats module is `@azure/agrifood-farming`.
+
+
+# How to generate DPG
+
+We are working on to automatically generate everything right now, but currently we still need some manual work to get a releasable package. Here're the steps of how to get the package.
+
+1. **Configure your tsp-location.yaml**  
+   
+   You can update `tsp-location.yaml` under project root folder to set the typespec project. You can refer to the [tsp-location.yaml](https://github.com/Azure/azure-sdk-tools/blob/main/doc/common/TypeSpec-Project-Scripts.md#tsp-locationyaml) which describes the supported properties in the file.
+   
+   If this is the first time to generate code you need to create the project folder.
+   ```shell
+   mkdir -p sdk/agrifood/agrifood-farming # create project folder
+   cd sdk/agrifood/agrifood-farming # enter project folder
+   ```
+   Then prepare the `tsp-location.yaml` and here is an example:
+
+   ```yaml
+   directory: specification/agrifood/DataPlane
+   commit: b646a42aa3b7a0ce488d05f1724827ea41d12cf1 # the commit id you'd like to refer for generation
+   repo: Azure/azure-rest-api-specs
+   ```
+   ---  
+    **NOTE**
+
+    We only allow to release any SDKs in which their TypeSpecs are merged into `Azure/azure-rest-api-specs` main branch so please ensure your TypeSpec pr is merged.
+
+   --- 
+
+2. **Generate code**
+
+    Run the following two scripts from project directory (i.e sdk/agrifood/agrifood-farming) to generate the code:
+
+    > NOTE: These scripts require PowerShell version 7 or higher
+    ```shell
+    pwsh ../../../eng/common/scripts/TypeSpec-Project-Sync.ps1 .
```

Comment:
@joheredi I may prefer to use sync script this time because I encoutered some issues in my local with command `tsp-client`. I need to report them with Catalina.

Reply:
In my initial doc @jeremymeng  noticed that I forgot to mention that they have to install the new tool.

Reply:
Yes @MaryGao please start recommending the `tsp-client` tool as it does the same as the powershell scripts. If you run into any issues report them so we can get them addressed quickly. The tool should also be much better for JS/TS developers since it comes from the same ecosystem, also your developers wont have to rely on having powershell.

Reply:
@catalinaperalta I met below issues could you help on it? After I manually removed that folder the issues fixed. Not sure why.

```
tsp-client version: 0.1.0
Found emitter package @azure-tools/typespec-ts
SDK initialized in C:\Users\marygao\project\azure-sdk-for-js\sdk\contentsafety\azure-ai-content-safety-rest
Additional directories: null
Error: fatal: destination path 'C:\Users\marygao\project\sparse-spec' already exists and is not an empty directory.
```

Reply:
I updated the doc to use `tsp-client` command.